hw/ip/usbdev/doc/wake_resume.md

```diff
@@ -13,24 +13,24 @@ The `usbdev_aon_wake` module maintains the pull-up state, monitors the D+ and D-
 
 ### Configuration
 
-0. Software configures usbdev to enable wake-from-deep-sleep on resume.
-   {{< regref "wake_config.wake_en" "hw/ip/usbdev/doc/_index.md" >}} should be set to `1`.
-1. Software configures the sleep state for usbdev DIOs to be high-Z (input only), in the `pinmux`.
+0. Software configures the sleep state for usbdev DIOs to be high-Z (input only), in the `pinmux`.
 
 ### Suspending
 
 1. The `usbdev_linkstate` module detects the line has been at idle for at least 3 ms, issues `link_suspend_o`, and transitions to the `LinkSuspended` state.
-   - The AON module transitions to the AwkTrigUon state and takes over the pullup enable.
-     It begins monitoring for events that trigger waking / resuming / resetting.
    - The protocol engine begins ignoring transactions.
      Any non-idle signaling kicks off the process to resume.
    - The USB differential receiver is turned off, if it was controlled by the `rx_enable_o` pin.
    - Software receives the event for going into the suspended state in {{< regref "intr_state.link_suspend" "hw/ip/usbdev/doc/_index.md" >}}.
-2. Software prepares for deep sleep.
+2. Software hands control to the AON module.
```

Comment:
do we want to clarify a bit when software hands control to the AON module?
Is this expected to happen in general? Or do you expect software to do this only when the system is about to engage a low power policy?

Reply:
also, when this hand-off occurs, do we need to do anything specific in the `usbdev` module itself to make sure it does not conflict with the always on wake detection module?

Reply:
Software only hands over control when it's about to engage a low power policy. If usbdev isn't going to be put in a low-power state, then software shouldn't activate the AON module.

Is the document's structure confusing? This particular line is an overview of the sub-bullets, and it would be performed after getting an interrupt for going into a suspended state, which is explained in the section just above. Were you looking for something more?

I believe nothing needs to be done in the `usbdev` module to avoid conflicts other than following the "Resuming" sequence below. The AON module reports its events and whether it is active to its own special CSR, and the only thing it actively takes over is the pull-up (which it keeps connected).

Reply:
sorry let me clarify a bit.

I think mainly it would be good to clarify that this should only be done when the system is ready to go into low power.
If for example we receive the suspend interrupt, but the system has no reason to go to low power yet (because some other thread is active), then we should not activate this sequence yet.  Is that already explained? I might have missed it. 

Regarding the last point, is there any worry that as that we're setting the wake module a resume comes in? Would that kick start the `usbdev` into doing something while the `aon_wake` module is still making its way through the first state? I'm wondering if we could start responding and then go silent (that's probably not a huge concern since we only have a handful of uS).  I'm also wondering in the event we are going into a clock shut off low power (ie, clocks are killed but power is not), if the hardware can get into an inconsistent state where both the aon_wake module and the usbdev are active.  Would that even be a problem?

Reply:
Ahh, the overview says it's a recipe for doing low-power modes. But I can add a blurb to confirm it isn't useful for the inverse.

There's no worry about when the resume comes in. The `aon_wake` module is complementary to `usbdev`. We make their operation windows overlap so no events are dropped, and the only requirement is that we can respond to an "Exit Suspended State" event within 10 ms from the point of no return. That would include powering down, powering back up, booting, restoring usbdev's state, and sending `wake_ack` to the `aon_wake` module.

Reply:
I added a bit to explain that this wouldn't be used without low-power mode and another that mentions what to do to resume before the point of no return.